core/modules/qana/PostPlugin.h

```diff
@@ -0,0 +1,68 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2016 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef LSST_QSERV_QANA_POSTPLUGIN_H
+#define LSST_QSERV_QANA_POSTPLUGIN_H
+
+// Qserv headers
+#include "qana/QueryPlugin.h"
+
+
+// Forward declarations
+namespace lsst {
+namespace qserv {
+namespace query {
+
+class OrderByClause;
+
+}}} // namespace lsst::qserv::query
+
+
+namespace lsst {
+namespace qserv {
+namespace qana {
+
+/// PostPlugin is a plugin handling query result post-processing.
+class PostPlugin : public QueryPlugin {
+public:
+    // Types
+    typedef std::shared_ptr<PostPlugin> Ptr;
+
+    virtual ~PostPlugin() {}
+
+    /// Prepare the plugin for a query
+    virtual void prepare() {}
+
+    /// Apply the plugin's actions to the parsed, but not planned query
+    virtual void applyLogical(query::SelectStmt&, query::QueryContext&);
+
+    /// Apply the plugins's actions to the concrete query plan.
+    virtual void applyPhysical(QueryPlugin::Plan& plan, query::QueryContext&);
+
+    int _limit;
+    std::shared_ptr<query::OrderByClause> _orderBy;
+};
+
+}}} // namespace lsst::qserv::qana
+
+
+#endif // LSST_QSERV_QANA_POSTPLUGIN_H
```

Comment:
If I understand correctly LSST C++ Style Guide does require the default contructor to be coded even if it's trivial. See: https://developer.lsst.io/coding/cpp_style_guide.html#constructors-and-destructors